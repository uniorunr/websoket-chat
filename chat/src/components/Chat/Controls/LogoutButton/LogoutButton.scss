@import "../../../../theme/colors";

.logout {
  display: flex;
  align-items: center;

  .logout__button-mobile {
    display: none;
    background: none;
    border: none;
    outline: none;
    cursor: pointer;
  }

  .logout__button {
    background: $black-color;
    color: $white-color;
    text-transform: uppercase;
    padding: 5px 15px;
    outline: none;
    cursor: pointer;
    border: 2px solid $black-color;
    border-radius: 5px;
  }

  .logout__button:hover {
    font-weight: 500;
    background: $white-color;
    color: $black-color;
    border-radius: 5px;
  }
}

@media only screen and (max-width : 549px) {
  .logout {
    .logout__button-mobile {
      display: flex;
    }

    .logout__button {
      display: none;
    }
  }
}

