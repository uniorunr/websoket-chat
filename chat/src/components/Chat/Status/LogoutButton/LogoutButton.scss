.logout-button {
  background: #000000;
  color: #ffffff;
  text-transform: uppercase;
  padding: 5px 15px;
  outline: none;
  cursor: pointer;
  border: 2px solid black;
  border-radius: 5px;
}

.logout-button:hover {
  font-weight: 500;
  background: #ffffff;
  color: #000000;
  border-radius: 5px;
}
