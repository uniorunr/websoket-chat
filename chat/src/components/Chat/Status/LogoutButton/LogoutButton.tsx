import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import * as actions from '../../../../actions/actions';
import './LogoutButton.scss';

const LogoutButton = ({ setUserName, updateAuthStatus }: LogoutButtonProps) => {
  const handleLogout = () => {
    setUserName('');
    updateAuthStatus(false);
    localStorage.removeItem('userName');
  };

  return (
    <button type="button" className="logout-button" onClick={handleLogout}>
      logout
    </button>
  );
};

interface LogoutButtonProps {
  setUserName: Function;
  updateAuthStatus: Function;
}

const mapDispatchToProps = (dispatch: Dispatch) => {
  const { setUserName, updateAuthStatus } = bindActionCreators(
    actions,
    dispatch
  );
  return {
    setUserName: (name: string) => {
      setUserName(name);
    },
    updateAuthStatus: (status: boolean) => {
      updateAuthStatus(status);
    }
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(LogoutButton);
