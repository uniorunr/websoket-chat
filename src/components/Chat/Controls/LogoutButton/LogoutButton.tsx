import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import * as actions from '../../../../actions/actions';
import LogoutMobileIcon from '../../../../assets/icons/logout.png';
import './LogoutButton.scss';

const LogoutButton = ({ setUserName, updateAuthStatus }: LogoutButtonProps) => {
  const handleLogout = () => {
    setUserName('');
    updateAuthStatus(false);
    localStorage.removeItem('userName');
  };

  return (
    <div className="logout">
      <button
        type="button"
        className="logout__button-mobile"
        onClick={handleLogout}
      >
        <img src={LogoutMobileIcon} alt="Logout Icon" width={20} />
      </button>
      <button type="button" className="logout__button" onClick={handleLogout}>
        logout
      </button>
    </div>
  );
};

interface LogoutButtonProps {
  setUserName: Function;
  updateAuthStatus: Function;
}

const mapDispatchToProps = (dispatch: Dispatch) => {
  const { setUserName, updateAuthStatus } = bindActionCreators(
    actions,
    dispatch
  );
  return {
    setUserName: (name: string) => {
      setUserName(name);
    },
    updateAuthStatus: (status: boolean) => {
      updateAuthStatus(status);
    }
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(LogoutButton);
