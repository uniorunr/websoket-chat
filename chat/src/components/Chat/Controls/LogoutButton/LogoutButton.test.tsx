import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../../store/mainStore';
import LogoutButton from './LogoutButton';

it('LogoutButton component renders correctly', () => {
  const app = shallow(
      <Provider store={store}>
        <LogoutButton />
      </Provider>
  );
  expect(app).toMatchSnapshot();
});
