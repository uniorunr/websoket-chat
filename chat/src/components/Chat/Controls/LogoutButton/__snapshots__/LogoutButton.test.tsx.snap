// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutButton component renders correctly 1`] = `
<div
  className="logout"
>
  <button
    className="logout__button-mobile"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Logout Icon"
      src="logout.png"
      width={20}
    />
  </button>
  <button
    className="logout__button"
    onClick={[Function]}
    type="button"
  >
    logout
  </button>
</div>
`;
